src/commands/synthetics/__tests__/reporters/default.test.ts

```diff
@@ -32,4 +37,324 @@ describe('Default reporter', () => {
+          ],
+          test: getNonBlockingApiTest('aaa-aaa-aaa'),
+        },
+      },
+    ]
+
+    test.each(cases)('$description', (testCase) => {
+      const {test, results, baseUrl, locationNames, failOnCriticalErrors, failOnTimeout} = testCase.fixtures
+      reporter.testEnd(test, results, baseUrl, locationNames, failOnCriticalErrors, failOnTimeout)
+      expect(writeMock).toHaveBeenCalledWith(testCase.expectedOutput)
+    })
+  })
+
+  describe('runEnd', () => {
+    beforeEach(() => {
+      writeMock.mockClear()
+    })
+
+    const resultsLabel = chalk.bold('Run summary:')
+    const testsNotFound = (n: number, publicIds: string[]) =>
+      `${chalk.yellow(`${chalk.bold(n)} test${n !== 1 ? 's' : ''} not found`)} ${chalk.gray(
+        `(${publicIds.join(', ')})`
+      )}`
+    const passed = (n: number) => chalk.green(`${chalk.bold(n)} passed`)
+    const failed = (n: number) => chalk.red(`${chalk.bold(n)} failed`)
+    const failedNonBlocking = (n: number) => chalk.yellow(`${chalk.bold(n)} failed (non-blocking)`)
+    const skipped = (n: number) => `${chalk.bold(n)} skipped`
+    const timedOut = (n: number) => chalk.yellow(`${chalk.bold(n)} timed out`)
+    const criticaErrors = (n: number) => chalk.red(`${chalk.bold(n)} critical errors`)
+
+    const baseSummary: Summary = createSummary()
+
+    const complexSummary: Summary = {
+      criticalErrors: 2,
+      failed: 1,
+      failedNonBlocking: 3,
+      passed: 2,
+      skipped: 1,
+      testsNotFound: new Set(['ccc-ccc-ccc', 'ddd-ddd-ddd']),
+      timedOut: 1,
+    }
+
+    const cases: {description: string; expectedOutput: string; summary: Summary}[] = [
+      {
+        description: 'Simple case with 1 test with 1 result (passed)',
+        expectedOutput: `${resultsLabel} ${passed(1)}, ${failed(0)}\n\n`,
+        summary: {...baseSummary, passed: 1},
+      },
+      {
+        description: 'Complex case with all the tests and results outcomes possible',
+        expectedOutput: [
+          testsNotFound(2, ['ccc-ccc-ccc', 'ddd-ddd-ddd']),
+          `${resultsLabel} ${passed(2)}, ${failed(1)}, ${failedNonBlocking(3)}, ${skipped(1)} (${timedOut(
+            1
+          )}, ${criticaErrors(2)})\n\n`,
+        ].join('\n'),
+        summary: complexSummary,
+      },
+      {
+        description: 'Case where some outcomes are empty or missing',
+        expectedOutput: [
+          testsNotFound(1, ['bbb-bbb-bbb']),
+          `${resultsLabel} ${passed(3)}, ${failed(0)}, ${failedNonBlocking(1)} (${criticaErrors(1)})\n\n`,
+        ].join('\n'),
+        summary: {
+          ...baseSummary,
+          criticalErrors: 1,
+          failedNonBlocking: 1,
+          passed: 3,
+          testsNotFound: new Set(['bbb-bbb-bbb']),
+        },
+      },
+    ]
+
+    test.each(cases)('$description', (testCase) => {
+      reporter.runEnd(testCase.summary)
+      expect(writeMock).toHaveBeenCalledWith(testCase.expectedOutput)
+    })
+  })
```

Comment:
[Jest snapshots](https://jestjs.io/docs/snapshot-testing) might prove useful for those tests, we don't use them in the synthetics reporter test suite but I think they could simplify things if you want to take a look at it